File: models/phrase.py
class Phrase:
    def __init__(self, phrase_id: str, english: str, japanese: str, romaji: str, category: str):
        self.phrase_id = phrase_id
        self.english = english
        self.japanese = japanese
        self.romaji = romaji
        self.category = category

    @classmethod
    def from_dict(cls, data: dict) -> "Phrase":
        return cls(
            phrase_id=data["id"],
            english=data["english"],
            japanese=data["japanese"],
            romaji=data["romanji"],
            category=data["category"],
        )

def matches_query(self, query: str) -> bool:
        q = query.lower()
        return (
            q in self.english.lower()
            or q in self.japanese.lower()
            or q in self.romaji.lower()
        )

File: flutter_app/lib/models/phrase.dart
class Phrase {
  final String id;
  final String categoryId;
  final String english;
  final String japanese;
  final String? romaji;
  final String? notes;
  final List<String> tags;

  const Phrase({
    required this.id,
    required this.categoryId,
    required this.english,
    required this.japanese,
    this.romaji,
    this.notes,
    this.tags = const [],
  });

  factory Phrase.fromMap(Map<String, dynamic> data) {
    final rawTags = data['tags'];
    final tags = (rawTags is List)
        ? rawTags.map((e) => e.toString()).toList(growable: false)
        : const <String>[];

    return Phrase(
      id: (data['id'] ?? '').toString(),
      categoryId: (data['category'] ?? data['categoryId'] ?? '').toString(),
      english: (data['english'] ?? '').toString(),
      japanese: (data['japanese'] ?? '').toString(),
      romaji: (data['romaji'] ?? data['romanji'])?.toString(),
      notes: data['notes']?.toString(),
      tags: tags,
    );
  }
}